addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicThingHandler.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    /**
+     * Converts the value of the datapoint to a State, updates the channel and also sets the thing status if necessary.
+     */
+    private void updateChannelState(final HmDatapoint dp, Channel channel)
+            throws IOException, BridgeHandlerNotAvailableException, ConverterException {
+
+        boolean isChannelLinked = isLinked(channel);
+        if (!dp.getChannel().isInitialized() && (isChannelLinked || dp.getChannel().getNumber() == 0)) {
+            synchronized (this) {
+                if (!dp.getChannel().isInitialized()) {
+                    try {
+                        getHomematicGateway().loadChannelValues(dp.getChannel());
+                    } catch (IOException ex) {
+                        if (dp.getChannel().getDevice().isOffline()) {
+                            logger.warn("Device '{}' is OFFLINE, can't update channel '{}'",
+                                    dp.getChannel().getDevice().getAddress(), channel.getUID());
+                        } else {
+                            throw ex;
+                        }
+                    }
+                }
+            }
+        }
+
+        boolean isStatusDatapoint = StringUtils.indexOfAny(dp.getName(), STATUS_DATAPOINT_NAMES) != -1;
+        if (isStatusDatapoint) {
+            updateStatus(dp.getChannel().getDevice());
+        }
+
+        if (isChannelLinked) {
+            TypeConverter<?> converter = ConverterFactory.createConverter(channel.getAcceptedItemType());
+            State state = converter.convertFromBinding(dp);
+            updateState(channel.getUID(), state);
+        }
+    }
+
+    /**
+     * Updates the thing status based on device status.
+     */
+    private void updateStatus(HmDevice device) {
+        ThingStatus oldStatus = thing.getStatus();
+        ThingStatus newStatus = ThingStatus.ONLINE;
+        ThingStatusDetail newDetail = ThingStatusDetail.NONE;
+
+        if (device.isFirmwareUpdating()) {
+            newStatus = ThingStatus.OFFLINE;
+            newDetail = ThingStatusDetail.FIRMWARE_UPDATING;
+        } else if (device.isUnreach()) {
+            newStatus = ThingStatus.OFFLINE;
+            newDetail = ThingStatusDetail.COMMUNICATION_ERROR;
+        } else if (device.isConfigPending()) {
+            newStatus = thing.getStatus();
+            newDetail = ThingStatusDetail.CONFIGURATION_PENDING;
+        }
+
+        if (thing.getStatus() != newStatus || thing.getStatusInfo().getStatusDetail() != newDetail) {
+            updateStatus(newStatus, newDetail);
+        }
+        if (oldStatus == ThingStatus.OFFLINE && newStatus == ThingStatus.ONLINE) {
+            initialize();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void updateStatus(ThingStatus status) {
+        super.updateStatus(status);
+    }
+
+    /**
+     * Returns true, if the channel is linked at least to one item.
+     */
+    private boolean isLinked(Channel channel) {
```

Comment:
please remove this method and use the existing BaseThingHandler.isLinked(channelId).